Add a few bits of additional context to address #30862 (#30923)
dotnet/docs: docs/standard/garbage-collection/implementing-disposeasync.md

### Unacceptable pattern

> The highlighted lines in the following code show what it means to have "stacked usings". If an exception is thrown from the `AnotherAsyncDisposable` constructor, neither object is properly disposed of. The variable `objTwo` is never assigned because the constructor did not complete successfully. As a result, the constructor for `AnotherAsyncDisposable` is responsible for disposing any resources allocated before it throws an exception. If the `ExampleAsyncDisposable` type has a finalizer, it is eligible for finalization. This problem is non-existent when using one of the acceptable patterns, as the clean up operations are correctly performed when they're not stacked.

Reviewer: ```suggestion
The highlighted lines in the following code show what it means to have "stacked usings". If an exception is thrown from the `AnotherAsyncDisposable` constructor, neither object is properly disposed of. The variable `objTwo` is never assigned because the constructor did not complete successfully. As a result, the constructor for `AnotherAsyncDisposable` is responsible for disposing any resources allocated before it throws an exception. If the `ExampleAsyncDisposable` type has a finalizer, it's eligible for finalization.
```

:::code language="csharp" id="dontdothis" source="snippets/dispose-async/ExamplePatterns.cs" highlight="9-10":::

> 

Reviewer: Then I'd put the bit about acceptable patterns as a second bullet point in the TIP block that starts on line 148.

```
- If you use one of the acceptable patterns, the problem of undisposed objects is non-existent. The clean-up operations are correctly performed when `using` statements aren't stacked.
```
